base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable = nameable.new())
+    super()
```

Comment:
- [ ] I received the attached error when I ran your program, and the error was about the default parameter you passed to the initialize method as an instance of the `Nameable` class, which is not supposed. Kindly remove the default value and pass `nameable` without the default value.

- [ ] Also, there is no use of the `super()` call you to have here too. Kindly remove it.

<img width="645" alt="chrome_IF8oZB8Wiu" src="https://user-images.githubusercontent.com/81468675/180480897-4cf6583a-8190-4d12-95e3-a6ba1391828b.png">